integration-cli/docker_cli_build_test.go

```diff
@@ -4941,6 +4941,40 @@ func (s *DockerSuite) TestBuildLabelsCache(c *check.C) {
 
 }
 
+func (s *DockerSuite) TestBuildNotVerbose(c *check.C) {
+	// This test makes sure that the `-s` mode works correctly
+	testRequires(c, DaemonIsLinux)
+	tt := []struct {
+		TestName     string // Silent Build Name
+		BuildCmds    string // Build string
+		ShouldFail   bool   // Expect to fail
+		StdoutRegexp string
+		StderrRegexp string
+	}{
+		{"testsilentbuild_busybox_and_echo", "FROM busybox\nRUN echo hello", false, "^(sha256:|)[a-z0-9]{64}\\n$", "^$"},
+		{"testsilentbuild_invalid_image", "FROM blabla", true, "^$", "Step 1 : FROM blabla((\\n|\r)+.*)+Error: image library/blabla(:latest)? not found\\n"},
+		{"testsilentbuild_exit_code_1", "FROM busybox\nRUN echo hello\nRUN exit 1", true, "^$", "Step 1 : FROM busybox((\\n|\r)+.*)+RUN echo hello((\\n|\r)+.*)+hello((\\n|\r)+.*)+Step 3 : RUN exit 1((\\n|\r)+.*)+The command '/bin/sh -c exit 1' returned a non-zero code: 1"},
```

Comment:
@duglin 
well. i am using `jsonmessage.DisplayJSONMessagesStream` at the daemon side. but as you can see, it is not that good: https://gist.github.com/ripcurld00d/b629c0f3473e0c831f16

my idea is to implement a function that detects the "[a-z0-9]{12}: [A-Z][a-z]+" pattern and shrink it. In that way, instead of getting:

```
a3ed95caeb02: Verifying Checksum
a3ed95caeb02: Download complete
6edcc89ed412: Verifying Checksum
6edcc89ed412: Download complete
6edcc89ed412: Pull complete
6edcc89ed412: Pull complete
a3ed95caeb02: Pull complete
a3ed95caeb02: Pull complete
```

the result will be:

```
6edcc89ed412: Pull complete
a3ed95caeb02: Pull complete
```

but i really don't like it. it's very ugly.

suggestions?